torch/distributed/tensor/_random.py

```diff
@@ -52,7 +52,10 @@ def manual_seed(seed: int, device_mesh: DeviceMesh) -> None:
 
     Args:
         seed (int): The desired seed.
-        device_mesh (:class:`DeviceMesh`): The device mesh to set the seed.
+        device_mesh (:class:`DeviceMesh`): The device mesh to set the seed. It is
+            required that the ``device_mesh`` include the calling rank. This is
```

Comment:
nit: "include the calling rank" is too weak a statement.  

It is required that 'device_mesh' contains the superset of all meshes that will be used with this DTensor.  It would be good to define 'spmd world' and then say that this mesh must be the spmd_world.